e2e/testcases/reconciler_manager_test.go

```diff
@@ -603,3 +603,37 @@ func TestAutopilotReconcilerAdjustment(t *testing.T) {
 	})
 
 }
+
+func TestAllowVerticalScaleResourceAdjustment(t *testing.T) {
+	nt := nomostest.New(t, ntopts.SkipMonoRepo)
+
+	reconcilerDeployment := &appsv1.Deployment{}
+	if err := nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem, reconcilerDeployment); err != nil {
+		nt.T.Fatal(err)
+	}
+	rs := &v1beta1.RootSync{}
+	if err := nt.Get(configsync.RootSyncName, configsync.ControllerNamespace, rs); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	_, err := nt.Kubectl("patch", "deployment", "reconciler-manager", "-n", "config-management-system", "-p", "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"args\":[\"--enable-leader-election\", \"--allow-vertical-scale\"],\"name\":\"reconciler-manager\"}]}}}}")
+	if err != nil {
+		nt.T.Fatalf("Error enabling vertical scale in deployment", err)
+	}
+
+	var updatedFirstContainerCPURequest int64
+	var updatedFirstContainerMemoryRequest int64
+	firstContainerName := reconcilerDeployment.Spec.Template.Spec.Containers[0].Name
+	updatedFirstContainerCPURequest = 159
+	updatedFirstContainerMemoryRequest = 259
```

Comment:
Would this pass in Autopilot clusters?

Reply:
the generation count could be different in autopilot cluster, i will add a condition to skip to separate the autopilot validation

Reply:
Do we need to skip in Autopilot? This should still work in Autopilot cluster. Maybe you can check whether a the CPU/memory request is changed by Autopilot, or just make your request compliant with Autopilot.

Reply:
added separate autopilot and regular cluster validation